### PR TITLE
cabana(DBCFile): handle escaped quotes

### PR DESCRIPTION
**Pull request by incognitojam**
**Description**
Add support for escaped quotemarks inside comments

**Verification**
- [x] Added test case
- [x] try opening/editing/saving with cabana

### CODE REVIEW

File: tools/cabana/tests/test_cabana.cc [hunk 1]
```diff
@@ -26,6 +26,15 @@ TEST_CASE("DBCFile::generateDBC") {
       REQUIRE(*sigs[i] == *new_sigs[i]);
     }
   }
+
+  QString content = R"(BO_ 160 message_1: 8 EON
+ SG_ signal_1 : 0|12@1+ (1,0) [0|4095] "unit" XXX
+
+CM_ BO_ 160 "message comment with \"escaped quotes\"";
```
Comment (sshane) [inline tools/cabana/tests/test_cabana.cc:33]: Does the DBC spec support this?

Comment (incognitojam) [reply tools/cabana/tests/test_cabana.cc:33]: Yeah this was from trying to open a leaked DBC saved with CanDB

**Comment by incognitojam** [comment]
I figured preserving the "header" (all of the definitions before the first message definition) would make opening/saving with cabana a bit less annoying too, although it still discards lots of other things.

**Comment by incognitojam** [comment]
CANParser doesn't like the empty val descriptions

**Comment by incognitojam** [comment]
Removed the empty val stuff, should be good to go

**Pull request merged**